### PR TITLE
refactor(cls): remove cls support

### LINKED ISSUE #10819: V6: Removing CLS support
We are currently considering remove CLS support as there are a few issues with it.

- It requires hooking the promise implementation which is not sustainable for the future of sequelize.
- It's generally unsafe due to it's implicit/magical nature, especially with developers that are unfamiliar with it's exact functionality.
- It isn't always reliable when mixed promise implementations were used.

Related issues:
- #9022
- #8978

We would like to gather some feedback/suggestions on this matter, we are currently looking at removing it entirely.

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

This is a little bit of a bigger breaking change I'd like to see in v6, this would also allow us to eventually ditch bluebird.

Closes #10819

### CODE REVIEW

File: docs/upgrade-to-v6.md [hunk 1]
```diff
@@ -7,3 +7,36 @@ Sequelize v6 is the next major release after v4
...
+### Removal of CLS
+
+CLS allowed implicit passing of the `transaction` property to query options inside of a transaction.
+This feature was removed for 3 reasons.
+
+- It required hooking the promise implementation which is not sustainable for the future of sequelize.
+- It's generally unsafe due to it's implicit nature.
+- It wasn't always reliable when mixed promise implementations were used.
+
+Check all your usage of the `.transaction` method and make you sure explicitly pass the `transaction` object for each subsequent query.
```
Comment (mirkojotic) [inline docs/upgrade-to-v6.md]: "make you sure" > "make sure you"?

File: appveyor.yml [hunk 2]
```diff
@@ -39,6 +39,7 @@ after_test:
 
 branches:
   only:
+    - v6
```
Comment (eseliger) [inline appveyor.yml]: this is duplicated

**Comment by fraywing** [comment]
Will there be a way for us to re-implement CLS without sequelize handling it? this is a pretty major breaking change if there is no migration path for those of us who have built whole apps around this. 

If you've wrapped most sequelize queries in functions in a very large app, it's a massive burden to pass `transaction` as an arg to each function signature.

**Comment by schmod** [comment]
Yeah.   This is in the "we're going to have to fork Sequelize" territory for us.

**Comment by papb** [comment]
@schmod See [my comment here](https://github.com/sequelize/sequelize/issues/10819#issuecomment-515262051). We will not remove CLS and just leave everyone hanging. Why is everyone so desperate? This PR was merged but v6 is still alpha.

**Comment by SimonSchick** [comment]
Correct, I just created a branch with the hooks required to implement cls as an external package (probably gonna call it sequelize-cls or something).
Part of v6 is to cut out/extract some features into separate components, sequelize has become quite monolithic which makes it hard to maintain at times.

**Comment by fraywing** [comment]
@SimonSchick Great to hear! Separate packages for this stuff makes a lot of sense.

**Comment by fraywing** [comment]
> @schmod See [my comment here](https://github.com/sequelize/sequelize/issues/10819#issuecomment-515262051). We will not remove CLS and just leave everyone hanging. Why is everyone so desperate? This PR was merged but v6 is still alpha.

I think it would be useful to mention this in releases. Everyone just looks at the release and assumes "welp, I guess I'm going to be SOL on the next major version bump"

**Pull request merged**